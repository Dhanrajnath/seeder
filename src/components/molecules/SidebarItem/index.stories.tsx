import { Meta, StoryObj } from "@storybook/react";
import SidebarItem, { ISidebarItemProps } from ".";
import HomeInActiveSvg from "../../../../public/assets/home-inactive.svg";
import CoinInActiveSvg from "../../../../public/assets/coin-inactive.svg";
import CoinActiveSvg from "../../../../public/assets/coin-active.svg";

export default {
    title: 'Molecules/SidebarItem',
    component: SidebarItem,
} as Meta;

type SidebarItemType = StoryObj<ISidebarItemProps>;


export const Home: SidebarItemType = {
    args: {
        itemName: "Home",
        iconPath: HomeInActiveSvg,
        isActive: false,
        onClick: () => {
            console.log("clicked sidebar item");
        }
    }
};

export const CashAcclerationInactive: SidebarItemType = {
    args: {
        itemName: "Cash Accleration",
        iconPath: CoinInActiveSvg,
        isActive: false,
        onClick: () => {
            console.log("clicked sidebar item");
        }
    }
};

export const CashAcclerationActive: SidebarItemType = {
    args: {
        itemName: "Cash Accleration",
        iconPath: CoinActiveSvg,
        isActive: true,
        onClick: () => {
            console.log("clicked sidebar item");
        }
    }
};