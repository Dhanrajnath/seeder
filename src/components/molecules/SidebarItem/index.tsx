import { Box, PaletteColor } from "@mui/material";
import IconFromSvg from "../../atoms/IconFromSvg";
import theme from "../../../utils/theme";
import Typography from "../../atoms/Typography";

export interface ISidebarItemProps {
    iconPath: string;
    itemName: string;
    isActive: boolean;
    onClick?: () => void
};

const SidebarItem = ({ iconPath, itemName, isActive, onClick }: ISidebarItemProps) => {
    return (
        <Box
            onClick={onClick}
            sx={{
                width: theme.spacing(52),
                height: theme.spacing(12),
                padding: theme.spacing(4),
                gap: theme.spacing(3),
                borderRadius: theme.spacing(3),
                background: isActive ? theme.palette.background.paper : theme.palette.background.default
            }}
            display="flex"
            flexDirection="row"
            alignItems="center"
            justifyContent="flex-start"
        >
            <IconFromSvg path={iconPath} alt={itemName} width={24} height={24} />
            <span
                style={{
                    color: isActive ? theme.palette.text.primary : theme.palette.text.disabled,
                    paddingLeft: theme.spacing(3)
                }}>
                <Typography variant="button">{itemName}</Typography>
            </span>
        </Box>
    );
};

export default SidebarItem;