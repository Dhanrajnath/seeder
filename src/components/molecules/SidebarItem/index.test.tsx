import { fireEvent, render } from "@testing-library/react";
import SidebarItem from ".";
import CoinActiveSvg from "../../../../public/assets/coin-active.svg";
import CoinInActiveSvg from "../../../../public/assets/coin-inactive.svg";

describe('SidebarItem test', () => {
    const mockOnClick = jest.fn();

    it('renders correctly', () => {
        const { container } = render(<SidebarItem iconPath={CoinActiveSvg} itemName="Cash accleration" isActive={true} onClick={mockOnClick} />);

        expect(container.firstChild).toMatchSnapshot();
    });
    it('renders sidebar item with active state', () => {
        const { getByText } = render(<SidebarItem iconPath={CoinActiveSvg} itemName="Cash accleration" isActive={true} onClick={mockOnClick} />);

        fireEvent.click(getByText('Cash accleration'));
        expect(mockOnClick).toHaveBeenCalled();
    });

    it('renders sidebar item with in-active state', () => {
        const mockOnClick = jest.fn();
        const { getByText } = render(<SidebarItem iconPath={CoinInActiveSvg} itemName="Cash accleration" isActive={false} onClick={mockOnClick} />);

        fireEvent.click(getByText('Cash accleration'));
        expect(mockOnClick).toHaveBeenCalled();
    });
});