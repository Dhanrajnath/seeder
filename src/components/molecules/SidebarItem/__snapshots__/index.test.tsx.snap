// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SidebarItem test renders correctly 1`] = `
<div
  class="MuiBox-root css-192sh95"
>
  <img
    alt="Cash accleration"
    height="24"
    src="coin-active.svg"
    width="24"
  />
  <span
    style="color: rgb(232, 231, 240); padding-left: 12px;"
  >
    <span
      class="MuiTypography-root MuiTypography-button css-13oj0q8-MuiTypography-root"
    >
      Cash accleration
    </span>
  </span>
</div>
`;
